src/test/java/com/ot/restaurant/entities/IoCContainerTest.java

```diff
@@ -0,0 +1,85 @@
+package com.ot.restaurant.entities;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class IoCContainerTest {
+
+    @Test
+    void singletonIoCContainer() {
+
+        IoCContainer ioc = IoCContainer.getInstance();
+        IoCContainer ioc2 = IoCContainer.getInstance();
+
+        assertEquals(ioc, ioc2);
+    }
+
+    @Test
+    void testRegisterAndResolve() throws Exception {
```

Comment:
Use a conventional naming for test such as:
```
void whenRegisterValueShouldResolveValue ()
```

The pattern is something similar to: `when_Should_` or `when_Then_` or `when_Must_`

Reply:
done